File: app/src/main/java/com/lostincontext/mainscreen/MainActivity.java
package com.lostincontext.mainscreen;

import android.os.Bundle;
import android.support.v4.app.FragmentManager;

import com.lostincontext.R;
import com.lostincontext.application.LostApplication;
import com.lostincontext.awareness.AwarenessModule;
import com.lostincontext.commons.BaseActivity;

import javax.inject.Inject;

import static com.lostincontext.utils.Activities.addFragmentToActivity;


public class MainActivity extends BaseActivity {

    @Inject MainScreenPresenter mainScreenPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main_screen);


        FragmentManager fm = getSupportFragmentManager();
        MainScreenFragment fragment = (MainScreenFragment) fm.findFragmentById(R.id.contentFrame);

        if (fragment == null) {
            fragment = MainScreenFragment.newInstance();
            addFragmentToActivity(fm, fragment, R.id.contentFrame);
        }

        // Create the presenter
        DaggerMainScreenComponent.builder()
                .mainScreenPresenterModule(new MainScreenPresenterModule(fragment))
                .applicationComponent(((LostApplication) getApplication()).getAppComponent())
                .awarenessModule(new AwarenessModule(this))
                .build()
                .inject(this);
    }


}

File: app/src/main/java/com/lostincontext/mainscreen/MainActivity.kt
package com.lostincontext.mainscreen

import android.os.Bundle
import com.lostincontext.R
import com.lostincontext.application.LostApplication
import com.lostincontext.commons.BaseActivity
import com.lostincontext.utils.addFragmentToActivity
import com.lostincontext.utils.logD


class MainActivity : BaseActivity() {

    val TAG = "MainActivity"


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main_screen)


        val fm = supportFragmentManager
        var fragment: MainScreenFragment? = fm.findFragmentById(R.id.contentFrame) as MainScreenFragment?

        if (fragment == null) {
            fragment = MainScreenFragment.newInstance()
            addFragmentToActivity(fm, fragment, R.id.contentFrame)
        }

        val lostApplication = application as LostApplication
        val userRepository = lostApplication.appComponent.userRepository
        userRepository.queryUsers("angie23", {
            logD(TAG) { it.body().toString() }
        }, { logD(TAG) { it.toString().toString() } })

    }


}
